content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/detect-log4shell-demo.md

```diff
@@ -0,0 +1,296 @@
+  53c080777a7d97samplegroupid07928aa3de055                  | log4demo  |                     
+  60adedf1d5db70examplerootidf32f6b7e09fc5                  | root      | Default user group  
+```
+
+Copy the ID value of the group managing the cluster you'll use to test Chainguard Enforce, and then use that value in place of `$COPIED_ID` in the following command. This will set an environment variable called `GROUP_ID` whose value is equal to the ID number Enforce uses to refer to your IAM group.
+
+```sh
+export GROUP_ID=$COPIED_ID
+```
+
+Next, run the following command to retrieve a list of all your active clusters. 
+
+```sh
+chainctl cluster list
+```
+```
+                  NAME                 |   GROUP   |               REMOTEID               | REGISTERED |    K8S VERSION    | AGENT VERSION | LAST SEEN |     ACTIVITY       
+---------------------------------------+-----------+--------------------------------------+------------+-------------------+---------------+-----------+--------------------
+  12345678-32fd-44b3-bea9-6046341ceda1 | log4demo  | 12345678-32fd-44b3-bea9-6046341ceda1 |      3m18s | v1.22.12-gke.2300 | 25bffd1-dirty |       12s | cosign-system:14s  
+                                       |           |                                      |            |                   |               |           |   eots-system:59s  
+                                       |           |                                      |            |                   |               |           | policy-system:12s
+
+```
+
+Notice the first column of information this command returns: the `NAME` column. The values in this column are the names that `chainctl` uses to manage each respective cluster.
+
+As you did with the IAM subgroup, copy the `NAME` value of the cluster you want to use for testing throughout this tutorial and use it in place of `$COPIED_NAME` to create an environment variable named `CLUSTER`.
+
+```sh
+export CLUSTER=$COPIED_NAME
+```
+
+
+## Step 2 – Running a Container Image Vulnerable to Log4Shell
+
+You're now ready to test how Chainguard Enforce works by deploying a sample application that includes a vulnerable version of Log4j. You'll then create a policy that will detect this vulnerability. 
+
+The sample container image used in this step is maintained by Chainguard for demonstration purposes only. Because this image includes a vulnerable version of Log4j, you **should not run this image in a production environment.** 
+
+If you'd like, you can inspect the image's [SBOM](https://edu.chainguard.dev/software-security/glossary/#sbom) by retrieving it directly from the container registry with the following `cosign` command. If you don’t already have `cosign` installed, you can install it following [these instructions from the Sigstore project documentation](https://docs.sigstore.dev/cosign/installation/).
+
+```sh
+cosign download sbom ghcr.io/chainguard-dev/log4shell-demo/app:v0.1.0
```

Comment:
Ah yeah, I caught that warning as I was working on this. 

To me, the warning is saying "Just because you're looking at this image's SBOM it doesn't mean that this image came from who you think it came from." But we've told the reader that this image comes from us, so this shouldn't be a concern for them.

I kept this command because I think it's a neat way to inspect an image's contents before you run it. If you think the warning is too off-putting I'd sooner just cut this command than replace it with the  `download attestation` command - that one returns a bunch of what looks like encrypted data and I'd rather not sidetrack things with a discussion about that stuff. Let me know!

Reply:
OK, can you add a phrase there just to let readers know that it allows you to inspect the image's content, and the warning is in place so that you can verify provenance.... and since we know where this is coming from we don't need to be concerned ...

Reply:
added! Thanks for following up